File: src/main/java/domisol/domain/member/entity/Member.java
package domisol.domain.member.entity;

import domisol.global.common.Status;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Entity
public class Member {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String socialId;

    private String email;

    private String nickname;

    @Enumerated(EnumType.STRING)
    private Status status;

    @Enumerated(EnumType.STRING)
    private Authority authority;

    @Builder
    public Member(String email, String nickname, String socialId, Status status, Authority authority) {
        this.email = email;
        this.nickname = nickname;
        this.socialId = socialId;
        this.status = status;
        this.authority = authority;
    }
}

File: flutter_/lib/models/member.dart
class Member {
  final int id;
  final String authority;
  final String email;
  final String nickname;
  final String profile_image;
  final String social_id;
  final String url;
  final String status;
  final DateTime created_at;
  final DateTime updated_at;

  Member({
    required this.id,
    required this.authority,
    required this.email,
    required this.nickname,
    required this.profile_image,
    required this.social_id,
    required this.url,
    required this.status,
    required this.created_at,
    required this.updated_at,
  });
}
